### PR TITLE
2716 cea plugin architecture

### PR DESCRIPTION
**Pull request by daren-thomas**
This PR implements the first stab at a plugin architecture for the CEA. It's still in its infancy but currently supported are:

- adding scripts to the CEA (CLI & GUI)
- adding new schemas to the CEA
- adding new plots to the CEA

A sample CEA plugin can be found in the [cea-plugin-template](https://github.com/architecture-building-systems/cea-plugin-template) repository. To install it, follow these steps:

- clone the repository in the CEA Console
- run `pip install -e .` in the plugin repo root (where the `setup.py` file is)
- run `cea-config write --general:plugins cea_plugin_template.DemandSummaryPlugin`
  - NOTE: maybe we should come up with a convention on how to add these plugins automatically when pip installing a cea plugin. I'll think about that. Could be a separate issue.

To test, try these:

- run `cea demand-summary` from CEA Console
- run Tools -> "Demand Summary" -> "Demand Summary" from the GUI
- add a plot ("Demand Summary" / "Total System Cooling Demand")

TODO:

- styling of the plots still needs some work
- demo some plot versions
- allow adding new parameters
- create a blog / howto / demo video for creating your own plugin (ideas for topics?)

### CODE REVIEW

File: cea/api.py [hunk 1]
```diff
@@ -11,14 +11,13 @@ def register_scripts():
...
+    config = cea.config.Configuration()
 
     def script_wrapper(cea_script):
         module_path = cea_script.module
         script_module = importlib.import_module(module_path)
 
-        def script_runner(config=None, **kwargs):
-            if config is None:
-                config = cea.config.Configuration()
+        def script_runner(config=config, **kwargs):
```
[the remark this answers is not in the document]
Comment (daren-thomas) [reply cea/api.py]: after scrum discussion, we're going to try putting it back again (we need config for the plugins) and we believe the issue in #2610 was due to the actual _saving_ of the configuration file at each load.

**Comment by reyery** [comment]
Nice! Now config loads almost instantly

**Comment by reyery** [comment]
Had to manually pip install cufflinks for plots in dashboard to work. Otherwise everything else works as expected.

**Comment by daren-thomas** [comment]
@reyery good catch: I updated the setup.py to include cufflinks.

**Pull request merged**